src/main/java/io/github/apace100/apoli/power/factory/condition/entity/PowerCondition.java

```diff
@@ -0,0 +1,41 @@
+package io.github.apace100.apoli.power.factory.condition.entity;
+
+import io.github.apace100.apoli.Apoli;
+import io.github.apace100.apoli.component.PowerHolderComponent;
+import io.github.apace100.apoli.data.ApoliDataTypes;
+import io.github.apace100.apoli.power.PowerType;
+import io.github.apace100.apoli.power.factory.condition.ConditionFactory;
+import io.github.apace100.calio.data.SerializableData;
+import io.github.apace100.calio.data.SerializableDataTypes;
+import net.minecraft.entity.Entity;
+import net.minecraft.util.Identifier;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PowerCondition {
+
+    public static boolean condition(SerializableData.Instance data, Entity entity) {
+
+        PowerType<?> powerType = data.get("power");
+        Identifier powerSource = data.get("source");
+
+        AtomicBoolean bl = new AtomicBoolean(false);
```

Comment:
Instead of using an `AtomicBoolean`, I'd prefer using `Optional#map` and then `orElse` here, e.g.
`PowerHolderComponent.KEY.maybeGet(entity).map(phc -> ...).orElse(false);`

I feel like that'd make the code easier to understand.